src/Process.php

```diff
@@ -110,13 +110,24 @@ public function start(LoopInterface $loop, $interval = 0.1)
             stream_set_blocking($pipe, 0);
         }
 
-        $loop->addPeriodicTimer($interval, function (Timer $timer) {
+        $periodicMonitor = function (Timer $timer) {
             if (!$this->isRunning()) {
                 $this->close();
                 $timer->cancel();
                 $this->emit('exit', array($this->getExitCode(), $this->getTermSignal()));
             }
-        });
+        };
+
+        $timer = $loop->addPeriodicTimer($interval, $periodicMonitor);
+
+        $dataMonitor = function ($data) use ($timer, $periodicMonitor) {
+            if ('' === $data) {
+                $periodicMonitor($timer);
+            }
+        };
+
+        $this->stdout->on('data', $dataMonitor);
+        $this->stderr->on('data', $dataMonitor);
```

Comment:
Re @WyriHaximus: These events could *possibly* be called very frequently - though the empty string check should not have a significant effect on performance.

Either way, we can probably avoid this by only binding on the `end` event here instead?